src/grpc/infer_handler.h

```diff
@@ -781,6 +911,11 @@ class InferHandlerState {
     std::queue<InferHandlerStateType*> states_;
     std::atomic<uint32_t> ongoing_requests_;
 
+    // Tracks the inflight requests sent to Triton core via this
+    // context. We will use this structure to issue cancellations
+    // on these requests.
+    std::set<InferHandlerStateType*> inflight_states_;
```

Comment:
By using a set, are we possibly hiding any bugs/errors where we insert a duplicate state, with the parallel nature of Handler/ResponseCallback/AsyncNotifyWhenDone logic for adding states onto queue?

Reply:
The items are added in this data structure iff the the InferAsync function was successfully called.